### PR TITLE
ceph_integration: Add ceph pool flows and atoms

### PR DESCRIPTION
**Pull request by cybernth**
Added the flows and atoms for pool CRUD

Signed-off-by: Nishanth Thomas <nthomas@nthomas.blr.redhat.com>

### CODE REVIEW

**Comment by cybernth** [comment]
@r0h4n @shtripat @brainfunked @nnDarshan Please review

File: etc/sds_operations_ceph-release-1-0.yaml [hunk 1]
```diff
@@ -1,436 +1,108 @@
...
-      pool.setomapval:
-        type: "update"
-      pool.rmomapkey:
-        type: "delete"
-      pool.getomapheader:
-        type: "get"
-      pool.setomapheader:
-        type: "update"
\ No newline at end of file
+namespace.tendrl.node_agent.ceph_integration:
```
Comment (shtripat) [inline etc/sds_operations_ceph-release-1-0.yaml]: remove node_agent

**Review by shtripat — APPROVED** [review]
1 comment

File: etc/sds_operations_ceph-release-1-0.yaml [hunk 2]
```diff
@@ -1,436 +1,108 @@
----
```
Comment (nnDarshan) [inline etc/sds_operations_ceph-release-1-0.yaml:1]: should this file name be renamed as tendrl_definitions_ceph_integration.yaml ?
@r0h4n

File: etc/sds_operations_ceph-release-1-0.yaml [hunk 3]
```diff
@@ -1,436 +1,108 @@
...
-      pool.rmomapkey:
-        type: "delete"
-      pool.getomapheader:
-        type: "get"
-      pool.setomapheader:
-        type: "update"
\ No newline at end of file
+namespace.tendrl.ceph_integration:
+  object_details:
+    Tendrl_context:
```
Comment (nnDarshan) [inline etc/sds_operations_ceph-release-1-0.yaml]: Isnt this part of node agent ?  should we have this definition here again ?

File: etc/sds_operations_ceph-release-1-0.yaml [hunk 4]
```diff
@@ -1,436 +1,108 @@
...
+              - Pool.pg_num
+              - Pool.pgp_num
+              - Pool.crush_ruleset
+            optional:
+              - Pool.max_objects
+              - Pool.max_bytes
+              - Pool.ec_profile
+          name: "Create Pool"
+          run: tendrl.ceph_integration.objects.Config.atoms.create_pool
+          type: Create
```
Comment (nnDarshan) [inline etc/sds_operations_ceph-release-1-0.yaml:51]: "Config" should start with a small "c" ?

Comment (nnDarshan) [reply etc/sds_operations_ceph-release-1-0.yaml:51]: this comment is for all the atoms

File: etc/sds_operations_ceph-release-1-0.yaml [hunk 5]
```diff
@@ -1,436 +1,108 @@
...
+        atoms:
+          - tendrl.ceph_integration.objects.Config.atoms.create_pool
+        description: "Create Ceph Pool"
+        enabled: True
+        inputs:
+            mandatory:
+              - "Pool[]"
+              - Tendrl_context.sds_name
+              - Tendrl_context.sds_version
+              - Tendrl_context.cluster_id
```
Comment (nnDarshan) [inline etc/sds_operations_ceph-release-1-0.yaml]: how does the create pool atom get other inputs like pool name, pool type , replica count etc ?

File: etc/sds_operations_ceph-release-1-0.yaml (hunk 3 above)
Comment (cybernth) [reply etc/sds_operations_ceph-release-1-0.yaml]: yes this can be done

File: etc/sds_operations_ceph-release-1-0.yaml (hunk 4 above)
Comment (cybernth) [reply etc/sds_operations_ceph-release-1-0.yaml:51]: ok

File: etc/sds_operations_ceph-release-1-0.yaml (hunk 5 above)
Comment (cybernth) [reply etc/sds_operations_ceph-release-1-0.yaml]: Its coming as part of job parameters. Means the pool[] will have parameters

File: etc/sds_operations_ceph-release-1-0.yaml (hunk 2 above)
Comment (shtripat) [reply etc/sds_operations_ceph-release-1-0.yaml:1]: Thats correct

File: etc/sds_operations_ceph-release-1-0.yaml [hunk 6]
```diff
@@ -1,436 +1,93 @@
...
-        type: "update"
-      pool.rmomapkey:
-        type: "delete"
-      pool.getomapheader:
-        type: "get"
-      pool.setomapheader:
-        type: "update"
\ No newline at end of file
+namespace.tendrl.ceph_integration:
+  object_details:
```
Comment (shtripat) [inline etc/sds_operations_ceph-release-1-0.yaml:2]: Rename to "objects"

File: etc/sds_operations_ceph-release-1-0.yaml [hunk 7]
```diff
@@ -1,436 +1,93 @@
...
+              - Pool.replica_count
+              - Pool.pg_num
+              - Pool.pgp_num
+              - Pool.crush_ruleset
+            optional:
+              - Pool.max_objects
+              - Pool.max_bytes
+              - Pool.ec_profile
+          name: "Create Pool"
+          run: tendrl.ceph_integration.objects.config.atoms.create_pool
```
Comment (shtripat) [inline etc/sds_operations_ceph-release-1-0.yaml:50]: There should be class name also mentioned here e.g.
tendrl.ceph_integration.objects.config.atoms.create_pool.CreatePool

**Pull request closed without merging**